src/compiler/proposals-and-stabilization.md

```diff
@@ -0,0 +1,336 @@
+  - **Propose using:** MCP
+  - **Approve using:** Seconding
+  - With the approval of the language team (that they think the feature is worth experimentation),
+    then submit an RFC and if, after discussion has concluded, a compiler team member agrees that
+    the implementation is feasible and will not put undue burden on the maintainers of the compiler,
+    then they can second the MCP and implementation can proceed.
+  - This isn't necessary if the owner of the implementation is a member of the compiler team
+- Stabilizing a language feature
+  - Follow the language team's process and have the stabiization PR reviewed by a member of the
+    compiler team
+- Reverting stabilization of a language feature
+  - Follow the language team's process and have the revert PR reviewed by a member of the
+    compiler team
+
+### Targets
+
+- Proposing a new target
+  - **Propose using:** PR
+  - **Approve using:** r+ (compiler leads)
+  - Open a PR with the new target (w/ relevant documentation updates) and document adherence to the
+    [target tier policy][tier_policy] in the description. New targets must start as tier three
+  - New targets should be assigned to the compiler team co-leads to check for any licensing
+    concerns
+- Promoting a target
+  - **Propose using:** PR
+  - **Approve using:** r+ (compiler leads)
+  - Open a PR with the new target and document adherence to the [target tier policy][tier_policy]
+    in the description
+  - New targets should be assigned to the compiler team co-leads to ensure that any demands on
+    the project infrastructure are considered and checked with relevant teams
+- Demoting/removing a target
+  - **Propose using:** MCP
+  - **Approve using:** FCP
+  - Write an MCP describing why the target should be demoted/removed and once discussion has
+    concluded, an FCP can be started to approve the demotion/removal.
+- Changing target baseline (e.g. minimum Darwin or Windows version bump)
+  - **Propose using:** MCP
+  - **Approve using:** FCP
+  - Write an MCP describing why the target should have a change of baseline and once discussion has
+    concluded, an FCP can be started to approve the change of baseline.
+- Adding/removing target maintainers
+  - **Propose using:** PR
+  - **Approve using:** r+
+  - Open a PR with the changes to the target documentation and obtain an r+ from the reviewer.
+- Adding a target feature
+  - **Propose using:** PR
+  - **Approve using:** r+
+  - Open a PR adding the target feature and obtain an r+ from the reviewer.
+- Stabilizing a target feature
+  - **Propose using:** PR
+  - **Approve using:** FCP
+  - Open a PR stabilizing the target feature and once the reviewer is happy with the changes,
+    an FCP can be started
+
+### Lints, errors and warnings
+
+- Adding a new warning/error
+  - **Propose using:** PR
+  - **Approve using:** r+
+  - Open a PR with the implementation and obtain an r+ from the reviewer
```

Comment:
Discussion: I think this has caveats in practice:

- If a new warning/error may hit a lot of crates (after crater run analysis), what's the reviewer supposed to do?
- If previously compiling code stopped compiling?

Reply:
I didn't add anything for these because I figured this was more of a reviewing concern than a approving-the-idea concern.

Reply:
Sure.